### PR TITLE
Support 5D tensors in the binary ops of Grappler Layout Optimizer (part 2)

### PR DESCRIPTION
**Pull request by kaixih**
This is to fix some issue found in https://github.com/tensorflow/tensorflow/pull/43166.

@andyly @nluehr

### CODE REVIEW

**Comment by andyly** [comment]
As https://github.com/tensorflow/tensorflow/pull/43166 was rolled back, can you rebase that PR into this one also?

**Comment by kaixih** [comment]
> As #43166 was rolled back, can you rebase that PR into this one also?

Sure. Rebased all the changes on top of master.

File: tensorflow/core/grappler/optimizers/generic_layout_optimizer_transposer.cc [hunk 1]
```diff
@@ -1093,16 +1122,18 @@ bool BinaryOpTransposer::IsFaninShapeSupported(
     const utils::MutableNodeView& node) {
   return (IsNDOperateWithMD(node, 4, 0) || IsNDOperateWithMD(node, 4, 1) ||
           IsNDOperateWithMD(node, 4, 4) || IsNDOperateWithMD(node, 0, 4) ||
-          IsNDOperateWithMD(node, 1, 4));
+          IsNDOperateWithMD(node, 1, 4) || IsNDOperateWithMD(node, 5, 0) ||
```
Comment (andyly) [inline tensorflow/core/grappler/optimizers/generic_layout_optimizer_transposer.cc:1125]: Can we also update `BinaryOpTransposer::IsFaninShapeSupported` to take in rank, to match `GetNDDataFaninPorts`?

**Review by andyly — CHANGES_REQUESTED** [review]

File: tensorflow/core/grappler/optimizers/generic_layout_optimizer_transposer.cc (hunk 1 above)
Comment (kaixih) [reply tensorflow/core/grappler/optimizers/generic_layout_optimizer_transposer.cc]: Yes, sounds good.

**Review by andyly — APPROVED** [review]

**Pull request merged**